services/spar/src/Spar/Data.hs

```diff
@@ -366,10 +372,14 @@ deleteIdPConfig
 deleteIdPConfig idp issuer team = retry x5 $ batch $ do
     setType BatchLogged
     setConsistency Quorum
+    addPrepQuery delDefaultIdp (Identity idp)
     addPrepQuery delIdp (Identity idp)
     addPrepQuery delIssuerIdp (Identity issuer)
     addPrepQuery delTeamIdp (team, idp)
   where
+    delDefaultIdp :: PrepQuery W (Identity SAML.IdPId) ()
+    delDefaultIdp = "DELETE FROM default_idp WHERE idp = ?"
```

Comment:
This is simple with the chosen schema and would be harder with a different primary key.